src/allotropy/parsers/agilent_gen5/agilent_gen5_structure.py

```diff
@@ -268,42 +268,65 @@ def create(cls, reader: LinesReader) -> ReadData:
         if read_type != ReadType.ENDPOINT:
             raise AllotropeConversionError(UNSUPPORTED_READ_TYPE_ERROR)
 
-        read_mode = cls.get_read_mode(procedure_details)
-        device_control_data = DeviceControlData.create(procedure_details, read_mode)
-        measurement_labels = cls._get_measurement_labels(device_control_data, read_mode)
-
-        number_of_averages = device_control_data.get(MEASUREMENTS_DATA_POINT_KEY)
-        read_height = device_control_data.get(READ_HEIGHT_KEY) or ""
-
-        return ReadData(
-            read_mode=read_mode,
-            step_label=device_control_data.step_label,
-            measurement_labels=measurement_labels,
-            detector_carriage_speed=device_control_data.get(READ_SPEED_KEY),
-            # Absorbance attributes
-            pathlength_correction=device_control_data.get(PATHLENGTH_CORRECTION_KEY),
-            number_of_averages=try_float_or_none(number_of_averages),
-            # Luminescence attributes
-            detector_distance=try_float_or_none(read_height.split(" ")[0]),
-            # Fluorescence attributes
-            filter_sets=cls._get_filter_sets(
-                measurement_labels, device_control_data, read_mode
-            ),
-        )
+        read_modes = cls.get_read_modes(procedure_details)
+        read_data_list: [ReadData] = []
+        full_read_lines: list[str] = procedure_details.splitlines()
+
+        section_lines_reader = SectionLinesReader(full_read_lines)
+        read_sections = []
+        for read_section in section_lines_reader.iter_sections("^Read\t"):
+            read_sections.append(read_section)
+
+        section = 0
+        for read_mode_idx in range(len(read_modes)):
+            read_mode = read_modes[read_mode_idx]
+            while section < len(read_sections):
```

Comment:
you dont need a loop here since it´s only being iterated once. One thing you can do is create a dict where its key is the read mode and the value is the red sections lines. This way you only need the initial loop on the read modes.

Reply:
Good suggestion - we ended up zipping the two lists together